locationsharinglib/locationsharinglib.py

```diff
@@ -272,26 +272,53 @@ def _get_data(self):
         url = 'https://www.google.com/maps/preview/locationsharing/read'
         response = self._session.get(url, params=payload)
         self._logger.debug(response.text)
+        return json.loads(response.text.split("'", 1)[1])
+
+    def get_shared_people(self):
+        """Retrieves all people that share their location with this account"""
         try:
-            output = json.loads(response.text.split("'", 1)[1])
+            output = self._get_data()
             people = [Person(info) for info in output[0]]
         except (IndexError, TypeError):
-            self._logger.exception('Response:%s', response.text)
-            return ()
+            self._logger.exception('Response: %s', output)
+            return []
         return people
 
+    def get_authenticated_person(self):
+        """Retrieves the person associated with this account"""
+        try:
+            output = self._get_data()
+            person = Person([
+                self.email,
+                output[9][1],
+                None,
+                None,
+                None,
+                None,
+                [
+                    None,
+                    None,
+                    self.email,
+                    self.email
+                ]
+            ])
+        except (IndexError, TypeError):
+            self._logger.exception('Response: %s', output)
+            return None
+        return person
+
     def get_all_people(self):
-        """Retrieves all people that share their location with this account"""
-        return self._get_data()
+        """Retrieves all people sharing their location"""
+        return filter(None, self.get_shared_people() + [self.get_authenticated_person()])
```

Comment:
Filter is not required and thus misleading, or am I missing something? You would just want to return the two lists as one so effectively the addition is enough.

Reply:
`get_authenticated_person()` can return None, and the filter seemed like a more elegant way of stripping it out.

Reply:
oh, right, sorry missed that on the small screen of my mobile. Looks great!